### DISCUSSION TITLE
Is SSR with hydration considered a performant approach?

### DISCUSSION THREAD

**Discussion by dariopog** [vercel/next.js#23534] [Help]
Hi everyone,

Is SSR with hydration considered a performant approach? From the article [Rendering on the Web](https://developers.google.com/web/updates/2019/02/rendering-on-the-web) on Google's blog _Web Fundamentals_, the authors say:

> Performance metrics collected from real websites using SSR rehydration indicate its use should be heavily discouraged.

Even if I fully understand the issues described in the article, I was a bit surprised to read this sentence and therefore I was wondering, despite such recommendation, if you still consider the use of SSR with hydration and Next.js a performant approach and what are your motivations.

Thx!

**Comment by dariopog** [comment]
No one? :)

**Comment by ekafyi** [comment]
There is a hacky solution here https://stackoverflow.com/questions/55393226/disable-hydration-only-partially-hydrate-a-next-js-app but yeah, I wish there was an easy way (eg. a config option) to disable hydration entirely ala SvelteKit for routes that don't need it.
